File: observer/Observer.java
import java.util.ArrayList;
import java.util.List;

interface Observer {
    void update(String message);
}

class Subscriber implements Observer {
    private String name;

    public Subscriber(String name) {
        this.name = name;
    }

    public void update(String message) {
        System.out.println("[" + name + "] Received update: " + message);
    }
}

class Publisher {
    private List<Observer> observers = new ArrayList<>();

    public void subscribe(Observer o) {
        observers.add(o);
    }

    public void unsubscribe(Observer o) {
        observers.remove(o);
    }

    public void notifyObservers(String message) {
        for (Observer o : observers) {
            o.update(message);
        }
    }

    public void publish(String message) {
        System.out.println("Publisher sends: " + message);
        notifyObservers(message);
    }
}

public class Main {
    public static void main(String[] args) {
        Subscriber alice = new Subscriber("Alice");
        Subscriber bob = new Subscriber("Bob");

        Publisher publisher = new Publisher();
        publisher.subscribe(alice);
        publisher.subscribe(bob);

        publisher.publish("Java 21 Released!");
    }
}

File: observer/Observer.py
from abc import ABC, abstractmethod

class Observer(ABC):
    @abstractmethod
    def update(self, message):
        pass

class Subscriber(Observer):
    def __init__(self, name):
        self.name = name

    def update(self, message):
        print(f"[{self.name}] Received update: {message}")

class Publisher:
    def __init__(self):
        self.observers = []

    def subscribe(self, observer):
        self.observers.append(observer)

    def unsubscribe(self, observer):
        self.observers.remove(observer)

    def notify_observers(self, message):
        for observer in self.observers:
            observer.update(message)

    def publish(self, message):
        print(f"Publisher sends: {message}")
        self.notify_observers(message)

# Example usage
alice = Subscriber("Alice")
bob = Subscriber("Bob")
publisher = Publisher()

publisher.subscribe(alice)
publisher.subscribe(bob)

publisher.publish("New Python tips released!")